src/node_types.ml

```diff
@@ -0,0 +1,34 @@
+type node_exports
+type node_module = <
+  id : string ;
+  exports : node_exports ;
+  parent : node_module Js.null_undefined;
+  (* in REPL V4 it is [undefined]
+    in CLI it can be [null]
+  *)
+
+  filename : string ;
+
+  loaded : Js.boolean;
+  children : node_module array ;
+  paths : string array;
+> Js.t
+ (** WARN:
+     its path name should not be changed
+     see {!Ppx_entry}
+*)
+
+type node_require = <
+  main : node_module Js.undefined;
+
+  resolve : string -> string [@bs]
+  (* @raise exception   *)
+> Js.t
```

Comment:
Sorry, this is my fault. I meant this just as a note for the future when you revamp the API. 

What I meant is that idiomatic OCaml does not typically use object types like those defined here. Instead you usually expose an abstract type and have a set of function to access and manipulate the contents of values of that type. That way you can change the internal representation of your data without it affecting your API.

This of course applies less to FFI, but you should still want your API to fit naturally in, and to be as intuitive as native OCaml APIs. So instead of this:
```ml
type foo = <
  id : string;
> Js.t
```
You should do this:
```ml
module Foo = struct
  type t
  external id : t -> string = "" [@@bs.get]
end
```

Could you revert your latest commit? Then I'll merge this PR and you can try to do this in the next one if you want.